### ISSUE TITLE
Support new CSP auth method for Wavefront (AOA)

### ISSUE THREAD

**Issue by oppegard** [micrometer-metrics/micrometer#4054]
**Please describe the feature request.**

Wavefront has integrated into VMware's [CSP](https://docs.vmware.com/en/VMware-Cloud-services/index.html) authentication system, and the `wavefront-sdk-java` library has been updated to support that. We would like micrometer to support this additional authentication method in `WavefrontMeterRegistry`

**Rationale**

New Wavefront customers will be onboarded to CSP-only clusters, and the existing authentication method of WF API Token will not be available to them. To allow these customers to use micrometer for sending metrics to Wavefront, they'll need CSP authentication available.

**Additional context**

The release version of `wavefront-sdk-java` with these changes is `3.4.0`.

There are two types of CSP authentication: server-to-server tokens and user-api-tokens. We envision the changes looking something like this:

- CSP User API Token
  - In `WavefrontMeterRegistry`, something like `new WavefrontClient.Builder(getWavefrontReportingUri(config), config.cspApiToken()).useTokenForCSP()`

- CSP Server to Server OAuth
  - In `WavefrontMeterRegistry`, something like `new WavefrontClient.Builder(getWavefrontReportingUri(config), config.cspClientId(), config.cspClientSecret())`
  - If an optional OrgID is provided, calling `builder.cspOrgId(config.cspOrgId())`

Note that CSP Authentication is only valid for Direct Ingestion scenarios (like `DEFAULT_DIRECT` in `WavefrontConfig`). It is not relevant for the `DEFAULT_PROXY` case.

Last, we've created a similar issue for Spring Boot: https://github.com/spring-projects/spring-boot/issues/37165

Thank you!

– @oppegard and @warshawd

**Comment by shakuzen** [comment]
Until dedicated support for this is added, users can provide a customized `WavefrontSender` to the `WavefrontMeterRegistery.builder()`. For example:

```java
WavefrontMeterRegistry meterRegistry = WavefrontMeterRegistry.builder(config)
    .wavefrontSender(WavefrontMeterRegistry.getDefaultSenderBuilder(config).useTokenForCSP().build())
    .build();
```

```java
WavefrontMeterRegistry meterRegistry = WavefrontMeterRegistry.builder(config)
    .wavefrontSender(new WavefrontClient.Builder(getWavefrontReportingUri(config), cspClientId, cspClientSecret)
        .cspOrgId(cspOrgId)
        .build())
    .build();
```

**Comment by shakuzen** [comment]
Resolved by #4102. A new configuration method `apiTokenType` has been added to `WavefrontConfig`. This, in conjunction with the existing `apiToken` method, can be used to configure the new CSP auth method. See https://github.com/wavefrontHQ/wavefront-sdk-java/pull/301 for more about the format to specify the token, especially when there are multiple parts.
